### PR TITLE
Add support for authentication codes via MDNS on iOS

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm [hunk 1]
```diff
@@ -98,15 +98,28 @@ - (void)publishServiceProtocolPort:(std::string)uri {
   NSString* serviceName =
       [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleIdentifier"];
 
+  // Check to see if there's an authentication code. If there is, we'll provide
+  // it as a txt record so flutter tools can establish a connection.
+  std::string path = std::string([[url path] UTF8String]);
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm]: Nit: `auto path = std::string{foo}` or `std::string path(foo)`. Repeating the type doesn't serve clarity in this case IMO.

File: shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm [hunk 2]
```diff
@@ -98,15 +98,28 @@ - (void)publishServiceProtocolPort:(std::string)uri {
...
+  if (!path.empty()) {
+    // Remove leading "/"
+    path = path.substr(1);
+  }
+  std::stringstream txtStr;
+  // txt records have the format [record length, byte0, byte1, ...]
+  txtStr << static_cast<char>(path.length()) << path;
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm]: Did you mean `std::to_string(path.length())`

Comment (chinmaygarde) [reply shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm]:
```
NSDictionary *txtDict = @{....};
NSData *txtData = [NSNetService dataFromTXTRecordDictionary:txtDict];
[netservice setTXTRecordData:txtData]; // or use the txtData.bytes
```

I believe that will manage encoding correctly. Let's just use that utility.

Comment (bkonyi) [reply shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm]: > Did you mean `std::to_string(path.length())`

Nope, wanted the byte to be equal to the length, not write the string representation to the stream.

> NSDictionary *txtDict = @{....};
NSData *txtData = [NSNetService dataFromTXTRecordDictionary:txtDict];
[netservice setTXTRecordData:txtData]; // or use the txtData.bytes
I believe that will manage encoding correctly. Let's just use that utility.

Done.

File: shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm (hunk 1 above)
Comment (bkonyi) [reply shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm]: Done.

**Pull request merged**